src/darwin/Framework/CHIP/MTRBaseDevice.mm

```diff
@@ -996,7 +996,13 @@ - (void)readAttributePaths:(NSArray<MTRAttributeRequestPath *> * _Nullable)attri
                 size_t count = 0;
                 eventPathParamsList
                     = static_cast<EventPathParams *>(Platform::MemoryCalloc([events count], sizeof(EventPathParams)));
-                VerifyOrReturnError(eventPathParamsList != nullptr, CHIP_ERROR_NO_MEMORY);
+                if (eventPathParamsList == nullptr) {
+                    if (attributePathParamsList != nullptr) {
+                        Platform::MemoryFree(attributePathParamsList);
```

Comment:
This is not enough.  You have to clean up both lists below if allocating `callback` or `readClient` fails.

That's why I suggested doing something RAII, because that makes it much harder to get things wrong and is a lot more maintainable.

Reply:
I modified using unique_ptr.
And, I have a question previous implementation.

Is it on purpose that the callback class memory was not erased from the subscription function?

When test005 was executed, it was found that memory delete was not called without onDone of the callback being called. (It is the same implementation even before I modified it.)

While changing to unique_ptr, additional modifications were made so that the callback class object memory was deleted, but if this part was done on purpose, I think it will need to be modified again.

Reply:
> Is it on purpose that the callback class memory was not erased from the subscription function?

What do you mean?

>  it was found that memory delete was not called without onDone of the callback being called

Well, until onDone is called the callback can be called into, no?  That's why onDone is the thing that deletes the callback...  Deleting it before then is not safe.

@joonhaengHeo I don't understand why we need to store the callback in the `MTRReadClientContainer`.  It should not need to be stored in there, as far as I can tell.  The setup for the callback in the subscribe case is:

1. It's stored in a unique_ptr on the stack.
2. If the subscribe is kicked off successfully, it guarantees that it will call onDone at some point.  So we `release()` the unique_ptr, and then in onDone we delete the callback.

Reply:
As long as onDone is guaranteed to be called, there should be no problem.
However, when I tested (test005_subscribe), I found a case where onDone was not called and ended, so I added these modifications.

Reply:
> I found a case where onDone was not called and ended

Was not called in what sense?  Was it not called when the subscription was actually terminated (i.e. when the controller got shut down) @joonhaengHeo ?  Or just not called within the test function, which is expected because the subscription does not terminate there....